File: apps/web/src/models/about.ts
import { z } from 'zod';

/**
 * Contract of `GET /api/about` exposed by the Go backend.
 *
 * This is the single source of truth for the application's product/creator
 * identity on the frontend: nothing in this codebase keeps a second copy of
 * the repository/creator/support URLs as a literal string - see
 * `internal/buildinfo` on the backend and `docs/product-identity-legal.md`.
 *
 * Display prose is intentionally NOT part of this contract - `versionLabel`-
 * style wording ("Development build") is derived from `isReleaseBuild` in
 * the UI layer via i18n, never sent as English text from the backend. The
 * licence fields are the one exception: a licence's own name/SPDX
 * identifier is not the kind of string that varies by UI language, so the
 * UI uses them directly rather than localizing them.
 */
export const aboutResponseSchema = z.object({
  productName: z.string().min(1),
  version: z.string().min(1),
  isReleaseBuild: z.boolean(),
  /** Short VCS revision, present only when Go's build-info stamping found one. */
  commit: z.string().min(1).optional(),
  commitDirty: z.boolean().optional(),
  creatorName: z.string().min(1),
  repositoryUrl: z.string().url(),
  creatorUrl: z.string().url(),
  supportUrl: z.string().url(),
  applicationLicenseSpdx: z.string().min(1),
  applicationLicenseName: z.string().min(1),
});

export type AboutResponse = z.infer<typeof aboutResponseSchema>;

File: apps/server/internal/httpapi/about.go
package httpapi

import (
	"log/slog"
	"net/http"

	"github.com/streaming-tree/server/internal/buildinfo"
)

// AboutResponse is the payload of GET /api/about.
//
// Every field here is either a fixed product-identity constant
// (internal/buildinfo) or build metadata Go can determine reliably on its
// own. Display prose ("Development build", licence status wording) is
// deliberately NOT sent as English text here - that is UI copy and belongs
// in the frontend's own localized strings, keyed off the status
// fields below.
//
// This response must never contain a Git author email, OS username,
// hostname, filesystem path, credential, token, or database path.
type AboutResponse struct {
	ProductName    string `json:"productName"`
	Version        string `json:"version"`
	IsReleaseBuild bool   `json:"isReleaseBuild"`
	// Commit and CommitDirty are omitted entirely when no VCS revision is
	// available, rather than sent as empty/false placeholders that could be
	// mistaken for a real (empty) commit.
	Commit                   string `json:"commit,omitempty"`
	CommitDirty              bool   `json:"commitDirty,omitempty"`
	CreatorName              string `json:"creatorName"`
	RepositoryURL            string `json:"repositoryUrl"`
	CreatorURL               string `json:"creatorUrl"`
	SupportURL               string `json:"supportUrl"`
	ApplicationLicenceStatus string `json:"applicationLicenceStatus"`
}

// aboutHandler reports fixed product-identity and build metadata for the
// About & Legal UI. It takes no dependencies beyond the logger: every field
// is a compile-time constant or something Go's own build-info stamping
// already knows, so unlike most routes in this package it needs no service
// interface and is always registered.
func aboutHandler(logger *slog.Logger) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		commit, dirty, ok := buildinfo.CommitInfo()

		resp := AboutResponse{
			ProductName:              buildinfo.ProductName,
			Version:                  buildinfo.Version,
			IsReleaseBuild:           buildinfo.IsReleaseBuild,
			CreatorName:              buildinfo.CreatorName,
			RepositoryURL:            buildinfo.RepositoryURL,
			CreatorURL:               buildinfo.CreatorURL,
			SupportURL:               buildinfo.SupportURL,
			ApplicationLicenceStatus: buildinfo.ApplicationLicenceStatus,
		}
		if ok {
			resp.Commit = commit
			resp.CommitDirty = dirty
		}

		writeJSON(w, logger, http.StatusOK, resp)
	}
}
